server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionService.java

```diff
@@ -740,6 +729,39 @@ private ConnectionEntity getEntity(@NonNull Long id) {
         return repository.findById(id).orElseThrow(() -> new NotFoundException(ResourceType.ODC_CONNECTION, "id", id));
     }
 
+    private List<ConnectionConfig> entitiesToModels(@NonNull List<ConnectionEntity> entities,
+            @NonNull Long organizationId, @NonNull Boolean withEnvironment) {
+        if (CollectionUtils.isEmpty(entities)) {
+            return Collections.EMPTY_LIST;
+        }
+        Map<Long, Environment> id2Environment;
+        if (withEnvironment) {
+            id2Environment = environmentService.list(organizationId).stream()
+                    .collect(Collectors.toMap(environment -> environment.getId(), environment -> environment));
+        } else {
+            id2Environment = new HashMap<>();
+        }
+        return entities.stream().map(entity -> {
+            ConnectionConfig connection = mapper.entityToModel(entity);
+            connection.setStatus(CheckState.of(ConnectionStatus.TESTING));
+            if (withEnvironment) {
+                Environment environment = id2Environment.getOrDefault(connection.getEnvironmentId(), null);
+                if (Objects.isNull(environment)) {
+                    return connection;
```

Comment:
return or throw exception?

Reply:
Yes, throwing an exception is a better way. Fixed.